### PR TITLE
threading related deprecation fixes for python 3.10

### LINKED ISSUE #1838: Threading related deprecations in Python 3.10
notifyAll has been deprecated in favor of notifyAll
currentThread has been deprecated in favor of current_thread
setDaemon has been deprecated in favor of setting daemon attribute

Reference : https://bugs.python.org/issue43723

```
➜  paramiko git:(master) rg -t py -w 'currentThread|notifyAll|activeCount|getName|setName|isDaemon|setDaemon'
demos/rforward.py
83:        thr.setDaemon(True)

paramiko/buffered_pipe.py
104:            self._cv.notifyAll()
206:            self._cv.notifyAll()

paramiko/util.py
235:    tid = id(threading.currentThread())

paramiko/sftp_file.py
530:        t.setDaemon(True)

paramiko/transport.py
420:        self.setDaemon(True)

paramiko/channel.py
1069:            self.out_buffer_cv.notifyAll()
1233:        self.out_buffer_cv.notifyAll()

tests/loop.py
84:            self.__cv.notifyAll()

tests/test_transport.py
747:                self.setDaemon(True)
771:                self.setDaemon(True)
```

### PR DESCRIPTION
**Pull request by Narendra-Neerukonda**
Fixes #1838

### CODE REVIEW

**Review by septatrix — APPROVED** [review]

**Comment by aliculPix4D** [comment]
@septatrix is there any chance this PR will be merged? Thanks.

**Comment by septatrix** [comment]
> @septatrix is there any chance this PR will be merged? Thanks.

I am not associated with this project so I have no control over this

**Comment by aliculPix4D** [comment]
> I am not associated with this project so I have no control over this

@septatrix sorry, I saw you as a reviewer so I automatically assumed you are responsible for merging it.

@bitprophet

**Comment by bskinn** [comment]
Are these changes backward compatible to Python 2.6 and 3.3?

If not, then I would assume they can't be merged as-is. Once @bitprophet finishes removing Python 2 support, that will no longer be an issue; and, I think he's planning to bump the minimum 3.x version at that same time, which will relax the backward compat constraint somewhat.

For now, though, >= 2.6 and >=3.3 are needed.


Separately, it's curious that the CI didn't run on this PR. Not sure what happened.

**Comment by bitprophet** [comment]
What @bskinn said - after next major release we'll support Python 3.6+ (and while we'll eventually edge closer to "matching upstream EOLs", I don't expect to drop 3.6 immediately either).

File: paramiko/sftp_file.py [hunk 1]
```diff
@@ -527,7 +527,7 @@ def _start_prefetch(self, chunks):
         self._prefetch_done = False
 
         t = threading.Thread(target=self._prefetch_thread, args=(chunks,))
```
Comment (graingert) [inline paramiko/sftp_file.py:529]:
```suggestion
        t = threading.Thread(target=self._prefetch_thread, args=(chunks,), daemon=True)
```

File: paramiko/sftp_file.py [hunk 2]
```diff
@@ -527,7 +527,7 @@ def _start_prefetch(self, chunks):
...
-        t.setDaemon(True)
+        t.daemon = True
```
Comment (graingert) [inline paramiko/sftp_file.py:530]:
```suggestion
```

File: paramiko/buffered_pipe.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def feed(self, data):
             if self._event is not None:
                 self._event.set()
             self._buffer_frombytes(b(data))
-            self._cv.notifyAll()
+            self._cv.notify_all()
```
Comment (bskinn) [inline paramiko/buffered_pipe.py:104]: To be sure I understand what's going on here -- it looks like you're updating a number of functions from deprecated CamelCase names to snake-case names, right?

If so -- then, you're sure that all of the snake-case names are available going all the way back to 2.6 & 3.0?

Comment (bskinn) [reply paramiko/buffered_pipe.py:104]: Ah, I missed the conversation that's occurring with @jun66j5, please ignore.

**Comment by bitprophet** [comment]
Cherry-picked to 2.9.x and up. Thanks all!

**Pull request closed without merging**